mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -746,6 +760,29 @@ impl MakerSwap {
         };
         swap_events.push(MakerSwapEvent::MakerPaymentInstructionsReceived(instructions));
 
+        let taker_amount = MmNumber::from(self.taker_amount.clone());
```

Comment:
I dont like that we have to clone BigDecimal amount in so many places to create MmNumber, while MmNumber From implementation actually uses reference
<img width="678" alt="image" src="https://github.com/user-attachments/assets/93eedecc-570a-424e-95e6-e6d99ad32a8d" />

<img width="787" alt="image" src="https://github.com/user-attachments/assets/99130b1d-9f8d-49b6-b61b-d230fddb0079" />


<img width="796" alt="image" src="https://github.com/user-attachments/assets/9def9088-d636-4a4a-94e8-3599433615f9" />

You dont need to do refactoring in this PR, I will leave it as issue to do https://github.com/KomodoPlatform/komodo-defi-framework/issues/2338#issue-2832592407